server/src/main/java/io/spine/server/aggregate/AggregateMessageEndpoint.java

```diff
@@ -51,17 +53,26 @@
     /**
      * Creates a tenant-aware operation based on the message this endpoint processes.
      */
-    abstract TenantAwareOperation createOperation();
+    abstract TenantAwareFunction0<R> createOperation();
 
     /**
      * {@linkplain #getTargets() Selects} one or more message targets and {@linkplain #dispatchTo(I)
      * dispatches} the message to them.
      */
-    void dispatch() {
-        final Set<I> targets = getTargets();
-        for (I target : targets) {
-            dispatchTo(target);
+    @SuppressWarnings("unchecked")
+    R dispatch() {
+        final R targets = getTargets();
+        if (targets instanceof Set) {
+            final Set<I> set = (Set<I>) targets;
+            for (I id : set) { //TODO:2017-07-21:alexander.yevsyukov: Can we run this in parallel?
+                //TODO:2017-07-21:alexander.yevsyukov: What if we dispatch events to multiple aggregates
```

Comment:
Then we should remember the issue and continue dispatching. And handle the exceptions later.

To be addressed outside of this PR.